### ISSUE TITLE
can I configure taffybar to be vertical?

### ISSUE THREAD

**Issue by laram1989** [travitch/taffybar#138]
Just set up this awesome piece of code with xmonad, but wondering if there is a configuration option to make it vertical (eg, to display on the right edge of my screen)? Unfortunately I really struggle to make senes of the Haskell code used in xmonad+taffybar, so my apologies if this is obvious... :(

**Comment by NerdGGuy** [comment]
There is no way to do this that I'm aware of. Three issues I can think of.

1. size is hard coded to be horizontal at top or bottom. Add Right and Left position options can fix this.
    setTaffybarSize :: TaffybarConfig -> Window -> IO ()

2. widgets stack horizontally, would need to change to vertical (GTK_ORIENTATION_VERTICAL).
    box <- hBoxNew False $ widgetSpacing cfg

3. widgets are horizontally orientated which may or may not be an issue for all widgets.

It'd be interesting if someone was willing to experiment with the first 2 and see what it looks like.

**Comment by travitch** [comment]
There is currently no way to do this.  The reasons noted by @NerdGGuy probably cover the bigger issues.  I have nothing against a vertical orientation, but haven't investigated at all.  The system tray might also be a bit of a challenge.  It is actually some older C code and might need to be modified to be vertical bar friendly.

**Comment by laram1989** [comment]
@NerdGGuy and @travitch, thank you for this, I suspected as much. On a laptop with a 16:9 screen vertical does seem very appealing, but regardless of that taffybar is great.

**Comment by saksham0808** [comment]
How do you propose to be able to see the window names, or the dbus notifications on a vertical bar? I'd love to experiment, but I don't see how's it going to handle all the text being vertical.

**Comment by NerdGGuy** [comment]
@saksham0808 Some of the widgets can still be rendered horizontally. Atleast that's how my Windows box use to be setup back in the day. I found vertical toolbars most useful in multimonitor setups when the toolbar could be made fairly wide on a single monitor.

**Comment by zearen** [comment]
Vertical support is the one thing stopping me from using TaffyBar.  I'm an experienced Haskell programmer.  Would you mind if I took a stab at this ?

**Comment by IvanMalison** [comment]
@zearen Of course not.

You may want to look at

https://github.com/IvanMalison/gtk-strut

which already supports vertical orientations, and:

https://github.com/taffybar/gtk-sni-tray

It really shouldn't be a ll that hard. I think the main issue will be that many of the widgets assume a certain orientation, so it will really be a matter of fixing each widget individually.

Now that I think about it, I think it would be a good idea to think about how we want the widgets to find out about their layouts. Let me know if you have any ideas.

**Comment by zearen** [comment]
Yeah, that was my evaluation.  It's just a bunch of busy work, which should be a nice break from my day job.

My first instinct was to add an `Orientation` enum to the widgets' constructor.  At worst, we could just ignore it with a `const`.

**Comment by IvanMalison** [comment]
@zearen All of the widgets are already in a reader monad that has context about the bar configuration and other things. I think it's probably better to put the information about the bar orientation there so we dont have to change the signature of every single widget.
